stable_baselines/common/distributions.py

```diff
@@ -293,12 +285,20 @@ def __init__(self, logits):
         :param logits: ([float]) the categorical logits input
         """
         self.logits = logits
+        self.action_mask = None
```

Comment:
Maybe some developers want to use Discrete.

So is this better?
class CategoricalProbabilityDistribution(ProbabilityDistribution):
    def __init__(self, logits, action_mask):

Reply:
Look at where the probability distribution is created ```proba_distribution_from_latent```, which is inherited from the base ```ProbabilityDistributionType```, which a few other classes use. To do this, you have to change the abstract method and add ```action_mask=None```. While it works, it isn't a great design pattern. The abstract class is being altered to allow for some child classes to have a feature, this is bad.

Reply:
With my approach, the base class is preserved, and there is no un-needed (inherent) passing of ```action_mask=None``` to the distributions that do not support action_masking.

Reply:
My original idea is that since we want to add action masks, all algorithms add action masks.🤔

In my version, model.step must do action_mask_check first. If action_mask=None, I will auto generate a blank action mask.

Reply:
that is not what the stable-baselines teams want to do: https://github.com/hill-a/stable-baselines/pull/453#issuecomment-538782084

Reply:
Can you check the latest commit for me?
I override proba_distribution_from_flat.
So, I restore the abstract class.

Reply:
I finally have a request, I want to contribute
Can I accept my PR and then overwrite my code?

Reply:
@H-Park,
Can you give me your e-mail address? Some things I want to talk to you, please.

Reply:
hpark@innovationdx.com